### PR TITLE
Parallel prompt tuning

### PR DESCRIPTION
**Pull request by vadam5**
# What does this PR do ?

This PR fixes prompt tuning to work with the updated pipeline parallel code. In this PR a user can also prompt tune using tensor parallel > 1.  More work will need to be done for prompt tuning to be supported for pipeline parallel > 1. 

**Collection**: [Note which collection this PR will affect]
BigNLP? 

# Changelog 
- prompt tuning with tp=1 and pp=1 works again with updated pipeline parallel code
- prompt tuning with tp > 1 works
- prompt tuning works with the updated complete and compute log probs methods
- Added a prompt tuning config file with prompting specific defaults set
- prompt tuning tests are add back

# Usage
```python
python examples/nlp/language_modeling/megatron_gpt_prompt_tuning.py \
    --config-name=megatron_prompt_tuning_gpt \
    restore_from_path='/prompt-tuning/megatron_gpt.nemo' \
    trainer.val_check_interval=2 \
    trainer.max_steps=5 \
    model.new_prompt_tags=['Winogrande, BoolQ'] \
    model.new_prompt_init_text=['logic choose person name, None'] \
    model.new_prompt_init_methods=['text, random'] \
    model.data.train_ds='/prompt-tuning/wino_bool_prompt_tuning_train.json' \
    model.data.valid_ds='/prompt-tuning/wino_bool_prompt_tuning_val.json' \
    +model.data.test_ds='/prompt-tuning/wino_bool_prompt_tuning_val.json' \
    model.micro_batch_size=2 \
    model.global_batch_size=4 \
    model.optim.lr=2e-2 \
    model.optim.sched.min_lr=2e-3 \
    model.optim.sched.warmup_steps=2 \
    model.optim.sched.constant_steps=8 \
    model.encoder_seq_length=2048
```

# Before your PR is "Ready for review"
**Pre checks**:
- [X] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [X] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
- [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [x] New Feature
- [X] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: nemo/collections/nlp/models/language_modeling/megatron_gpt_model.py [hunk 1]
```diff
@@ -14,6 +14,7 @@
 
 import os
 import re
+from tkinter import W
```
Comment (ericharper) [inline nemo/collections/nlp/models/language_modeling/megatron_gpt_model.py:17]: I don't think you meant to import this?

**Review by ericharper — CHANGES_REQUESTED** [review]
I think there's a couple of imports that you did not want to add, see the comments.

**Review by ericharper — APPROVED** [review]
LGTM. Thanks!

**Review by okuchaiev — APPROVED** [review]
Thanks!

**Pull request merged**